### ISSUE TITLE
Editor find input controls are visually grouped but not programmatically associated / focus order inconsistent

### ISSUE THREAD

**Issue by jjaeggli** [microsoft/vscode#167885]
Controls within the editor find dialog corresponding to the find input are visually grouped but not programmatically associated with one another. The "Match case", "Match word", and "Match regular expression" checkboxes are visually located within the input field. However, within the accessibility tree, they are not grouped, therefore there is no direct association between these controls and the "Find" input.

![Screen Shot 2022-11-29 at 10 35 26 AM](https://user-images.githubusercontent.com/13710580/205163712-d4ce140a-5eb1-45dc-88e5-d0a076972b10.png)

This is additionally complicated when the "Replace" toggle is expanded. When this occurs, the focus order of the controls change, and does not follow the visual structure of the page.

![Screen Shot 2022-12-01 at 2 38 04 PM](https://user-images.githubusercontent.com/13710580/205164668-2bfc8465-2948-4e87-befe-6cac38dd2eb9.png)

WCAG 2.1 1.3.1 Level A specifies that visual relationships perceivable by one set of users is preserved for all users. This can be achieved by [grouping related form controls](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html). 

WCAG 2.4.3 specifies that focus order should allow a user to navigate through the content in a sequential and consistent order that reflects the logical relationships present on the page. When the "Replace" toggle is expanded, the focus order of the elements change so that the find input controls now occur after the replace input field, and the "Preserve Case" control follows this, with no indication that "Preserve Case" is a new control which relates to replace.

My recommendation is that find and replace toggles are programmatically grouped with their associated fields, and that focus order respects the visual grouping of these controls.

Persistent visual labels provided by FR #167870 can be used to label the groups.

**Comment by jjaeggli** [comment]
I wanted to see if any discussion had taken place in regard to changing the focus order of these controls. I understand that the current order when replace is expanded is intentional to allow users to quickly navigate between find and replace is intentional, but changing the focus order is not ideal, as focus should remain consistent and predictable. I would be happy to send a PR for this change, but I want to get a consensus on the solution first.

**Comment by amunger** [comment]
>  I understand that the current order when replace is expanded is intentional to allow users to quickly navigate between find and replace is intentional

Correct, we wouldn't want to force extra tabs in order to get to switch back and forth between those two fields.

It seems like a UI re-organization would be required to actually fix this.
